x-pack/elastic-agent/pkg/core/plugin/process/start.go

```diff
@@ -39,7 +39,7 @@ func (a *Application) start(ctx context.Context, t app.Taggable, cfg map[string]
 	}()
 
 	// already started if not stopped or crashed
-	if a.state.Status != state.Stopped && a.state.Status != state.Crashed && a.state.Status != state.Failed {
+	if a.Started() {
```

Comment:
The internal `start()` is called to restart on a crash. We sure this change with the above change is not going to affect that behavior?

Reply:
this should be ok we check it only 2 places here (which stays unaffected, it just looks more readable) 
and operation_start.Check which is called on config resolution